src/introductory-offer-banner/index.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * External dependencies
+ */
+import { Fill, Card, Button } from '@wordpress/components';
+import { __, sprintf } from '@wordpress/i18n';
+import { OPTIONS_STORE_NAME } from '@woocommerce/data';
+import { useSelect, useDispatch } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+
+export const WC_CALYPSO_BRIDGE_INTRODUCTORY_OFFER_BANNER_HIDDEN =
+	'wc_calypso_bridge_introductory_offer_banner_hidden';
+
+export const CalypsoBridgeIntroductoryOfferBanner = () => {
+	const { updateOptions } = useDispatch( OPTIONS_STORE_NAME );
+	const { isModalHidden } = useSelect( ( select ) => {
+		const { getOption, hasFinishedResolution } =
+			select( OPTIONS_STORE_NAME );
+
+		return {
+			isModalHidden:
+				getOption(
+					WC_CALYPSO_BRIDGE_INTRODUCTORY_OFFER_BANNER_HIDDEN
+				) === 'yes' ||
+				! hasFinishedResolution( 'getOption', [
+					WC_CALYPSO_BRIDGE_INTRODUCTORY_OFFER_BANNER_HIDDEN,
+				] ),
+		};
+	} );
+
+	const dismissModal = () => {
+		updateOptions( {
+			[ WC_CALYPSO_BRIDGE_INTRODUCTORY_OFFER_BANNER_HIDDEN ]: 'yes',
+		} );
+	};
+
+	return (
+		! isModalHidden && (
+			<Fill name="woocommerce_homescreen_experimental_header_banner_item">
+				<Card>
+					<div className="wc-calypso-bridge-woocommerce-admin-introductory-offer-banner">
+						<div className="wc-calypso-bridge-woocommerce-admin-introductory-offer-banner__text">
+							<p>
+								{ __(
+									'Start selling — for less!',
+									'wc-calypso-bridge'
+								) }
+							</p>
+							<p>
+								{ sprintf(
+									/* translators: First two %s do not need to be translated. Possible values for the last %s is day, days, month, and months */
+									__(
+										'Upgrade your plan for %s for your first %s %s',
```

Comment:
Sorry to be a bearer of bad news, but I think the handling for the copy here is not exactly straightforward 😬 

[Here's how it's done in calypso](https://github.com/Automattic/wp-calypso/blob/5b6338f8ec5fea0d9bf70fb317f0db9ab1a221ff/client/my-sites/plans-grid/components/billing-timeframe.tsx#L96-L197), tldr is that we may need to handle different intervals unit & count and as separate translations. Hopefully we could copy and adapt some of the calypso code to perform the logic here.

Reply:
@moon0326  Good point, but I think we still need to handle this slightly more robust. I think we still need to still translate the period, such as `day`, `month`, `year`, along with the plural variations.

ChatGPT provided this logic 😆 :

```js
let message = '';
if (intervalCount === '1') {
    switch (formattedIntervalUnit) {
        case 'day':
            message = sprintf(__('Upgrade your plan for %s for your first day', 'wc-calypso-bridge'), formattedPrice);
            break;
        case 'month':
            message = sprintf(__('Upgrade your plan for %s for your first month', 'wc-calypso-bridge'), formattedPrice);
            break;
        case 'year':
            message = sprintf(__('Upgrade your plan for %s for your first year', 'wc-calypso-bridge'), formattedPrice);
            break;
    }
} else {
    switch (formattedIntervalUnit) {
        case 'days':
            message = sprintf(__('Upgrade your plan for %s for your first %s days', 'wc-calypso-bridge'), formattedPrice, intervalCount);
            break;
        case 'months':
            message = sprintf(__('Upgrade your plan for %s for your first %s months', 'wc-calypso-bridge'), formattedPrice, intervalCount);
            break;
        case 'years':
            message = sprintf(__('Upgrade your plan for %s for your first %s years', 'wc-calypso-bridge'), formattedPrice, intervalCount);
            break;
    }
}
```

Reply:
Ah! I see what you mean @ilyasfoo Thank you!

I'll update it soon.

Reply:
Updated in https://github.com/Automattic/wc-calypso-bridge/pull/1397/commits/c36fd20d9d80cf58f377ce58a748300b4c851d9e

@ilyasfoo